### ISSUE TITLE
Question of dynamic node classification & edge classification?

### ISSUE THREAD

**Issue by sherjy** [yule-BUAA/DyGLib#3]
Hii Yu, thanks a lot for your nice work and codes sharing!! I have a question about the dynamic node classification, after reading the codes, I found in the `DG_data/DATASETS_README.md` that it says the title of label column is the `edge_label `, but indeed it's the dynamic node classification task of `src_node` (shown in `train_node_classification.py` file). So there's a minor misleading issue here. Moreover, do you have plan to support edge classification in DyGLib? Thanks in advance! ^0^

**Comment by yule-BUAA** [comment]
Hi,

Thanks for your appreciation of our work! ^_^

### Name of edge_label
Your understanding of `edge_label` is correct! In fact, we use the same column names as previous works do. We name node labels as `edge_label` because, in each interaction (i.e., edge), the involved source node is associated with a label, which means **the number of node labels is equal to the edge number**. In scenarios where the node tends to be assigned with a label **after a number of interactions**, I think using `node_label` is more appropriate. But it will be also fine to replace `edge_label` with `node_label` if you can understand better from the latter! 

### Edge classification task
Actually, the current link prediction task is a binary classification task for edges (with label 0 or 1) since most existing datasets support such a setting. If your datasets have more than two edge label classes, you can simply do the following things to achieve multi-class edge classification. 
1. adjust the loss function from `BCELoss` to `CrossEntropyLoss` [here](https://github.com/yule-BUAA/DyGLib/blob/fcb9721a48ed6904f4139e660402713661744edc/train_link_prediction.py#L144);
2. set the `output_dim` of MergeLayer to the number of edge label classes [here](https://github.com/yule-BUAA/DyGLib/blob/fcb9721a48ed6904f4139e660402713661744edc/train_link_prediction.py#L126).

Please feel free to contact me if there are any further questions.

**Comment by sherjy** [comment]
Thanks a lot for quick reply! The first question of `edge_label` has been well clarified. There's a concern in second Q of `edge classification`. Since the future link prediction usually uses concat(dynamic src_node embedding, dynamic dst_node embedding) to predict whether an interaction occurs at target timestamp, the edge classification uses concat(dynamic src_node embedding, temporal interaction feature, dynamic dst_node embedding) to classify the label of target edge. So I wanna know there's a small gap of these two tasks right? (If I want to modify into an edge classification task, I need do some small modifications basic on `node_classification` better. Correct me if I'm wrong.) Thanks for clarification!

**Comment by yule-BUAA** [comment]
Based on the current implementation, it is easier to realize edge classification using the link prediction code. It is a convenient extension to combine temporal interaction features with dynamic src_node embedding and dynamic dst_node embedding. And the above two modifications I mentioned are also straightforward to do.

Currently, the node classification code is unsuitable since its pipeline is first using the pre-trained temporal GNN models (derived from the link prediction task, and **the parameters are freezed**) to obtain node embeddings and then only finetuning the classifier. Such a design is used to tackle the _imbalanced node labels_ problem following previous works, but this issue is unlikely to happen on the edge classification task.

**Comment by sherjy** [comment]
Thanks for the clarification and it's all clear now.

**Comment by zhiqiuyuan** [comment]
> Based on the current implementation, it is easier to realize edge classification using the link prediction code. It is a convenient extension to combine temporal interaction features with dynamic src_node embedding and dynamic dst_node embedding. And the above two modifications I mentioned are also straightforward to do.
> 
> Currently, the node classification code is unsuitable since its pipeline is first using the pre-trained temporal GNN models (derived from the link prediction task, and **the parameters are freezed**) to obtain node embeddings and then only finetuning the classifier. Such a design is used to tackle the _imbalanced node labels_ problem following previous works, but this issue is unlikely to happen on the edge classification task.

Hi, 
I'm interested in "Such a design is used to tackle the imbalanced node labels problem following previous works",
Could you please tell me what the "previous works" are? I want to take a look at them! Thanks!
